admin/class-acc_user_importer-admin.php

```diff
@@ -315,191 +326,234 @@ private function proccess_user_data ( $users ) {
+					$update_this_user = true;
+				}
+
+				//Check if MEMBERSHIP_N changed
+				if ($userMembership != $user_meta->membership) {
+					$this->log_dual(" > membership# changed from " . $user_meta->membership . " to " . $userMembership);
+					$update_this_user = true;
+				}
+
+				//Check if Membership Expiry changed
+				if ($userExpiry != $user_meta->expiry) {
+					$this->log_dual(" > expiry changed from " . $user_meta->expiry . " to " . $userExpiry);
+					$update_this_user = true;
+				}
+
+				//Check if city changed
+				if ($userCity != $user_meta->city) {
+					$this->log_dual(" > city changed from " . $user_meta->city . " to " . $userCity);
+					$update_this_user = true;
+				}
+
+				//Introduce a special rule to NOT update a user if the incoming data has
+				//a expiry date earlier than the one in the local DB. This is because
+				//sometimes a user has 2 memberships, one family and one personal, with
+				//different information in each. When the plugin runs, it receives asynchronously
+				//the 2 memberships, so one overwrites the other. Which one is the best one
+				//is hard to say, but most likely the information in the membership with
+				//latest expiry date is the best, because it is the latest one subscribed
+				//to by the user.
+				//I think we can do a straight string compare, given the YYYY-MM-DD-TIME format.
+				if ($userExpiry < $user_meta->expiry) {
+					$this->log_dual(" > Received expiry is earlier than expected. Reject update");
 					$update_this_user = false;
-					$update_errors[] = $user;
 				}
-				
-				//no duplicate email exists
-				else {
-					$api_response['log'] .= "&nbsp;&gt; email not found on any other users<br/>";
-					$api_response['log'] .= "&nbsp;&gt; creating new user account<br/>";
-					$this->log_local_output(" > email not found on any other users");
-					$this->log_local_output(" > creating new user account");
-					$new_users[] = $user_contact_id;
+
+				if (!$update_this_user) {
+					$this->log_dual(" > Nothing changed for this user");
+				}
+
+			//--------USER NOT FOUND IN DATABASE-----
+			//But before creating a new record, make sure email is unique.
+			//We want emails to be unique in DB because it is a login identifier.
+			} elseif ( !(strlen($userEmail) > 1) ) {
+				//User has no email field, skip it
+				$this->log_dual(" > error: no email given, cannot create new user account.");
+				$update_errors[] = $user;
+
+			} else {
+				$this->log_dual(" > user not found");
+
+				$user_id = email_exists($userEmail);
+				if ( is_numeric($user_id) ) {
+					//An existing user already has this email address, skip updating
+					$user2 = get_userdata($user_id);
+					$username2 = $user2->user_firstname . " " . $user2->user_lastname;
+					$this->log_dual(" > existing user #" . $user_id . " " . $username2 .
+					                " already has email " . $userEmail);
+					$this->log_dual(" > user update skipped");
```

Comment:
My understanding is that this is not always correct. If `update_this_user` is set, we still might update it, right?

Reply:
The only way to get here is with update_this_user being false. So we will not update it.